doc/source/tutorial/interpolate.rst

```diff
@@ -10,38 +10,494 @@ Interpolation (:mod:`scipy.interpolate`)
+``len(t) - len(x) - k - 1`` boundary conditions. For cubic splines with
+``(k+1)``-regular knot arrays this means two boundary conditions---or
+removing two values from the ``x`` array. Various boundary conditions can be
+requested using the optional ``bc_type`` argument of `make_interp_spline`.
+
+The b-spline coefficients are accessed via the ``c`` attribute of a `BSpline`
+object:
+
+    >>> len(bspl.c)
+    7
+
+The convention is that for ``len(t)`` knots there are ``len(t) - k - 1``
+coefficients. Some routines (see the :ref:`Smoothing splines section
+<tutorial-interpolate_fitpack>`) zero-pad the ``c`` arrays so that
+``len(c) == len(t)``. These additional coefficients are ignored for evaluation.
+
+Also note that the coeffients are given in the
+:ref:`b-spline basis <tutorial-interpolate_bspl_basis>`, not the power basis
+of :math:`1, x, \cdots, x^k`.
+
+
+.. _tutorial-interpolate_bspl_basis:
+
+B-spline basis elements
+-----------------------
+
+B-splines are piecewise polynomials, represented as linear combinations of
+*b-spline basis elements* --- which themselves are certain linear combinations
+of usual monomials, :math:`x^m` with :math:`m=0, 1, \dots, k`.
+
+The b-spline basis generally more computationally stable then the power basis
+and is useful for variety of applications which include interpolation, regression
+and curve representation. The main feature is that these basis elements are
+*localized* and equal zero outside of an interval defined by the *knot array*.
+
+Specifically, a b-spline basis element of degree ``k`` (e.g. ``k=3`` for cubics)
+is defined by :math:`k+2` knots and is zero outside of these knots.
+To illustrate, plot a collection of non-zero basis elements on a certain
+interval:
+
+.. plot ::
+
+    >>> k = 3      # cubic splines
+    >>> t = [0., 1.4, 2., 3.1, 5.]   # internal knots
+    >>> t = np.r_[[0]*k, t, [5]*k]   # add boundary knots
+
+    >>> from scipy.interpolate import BSpline
+    >>> import matplotlib.pyplot as plt
+    >>> for j in [-2, -1, 0, 1, 2]:
+    ...     a, b = t[k+j], t[-k+j-1]
+    ...     xx = np.linspace(a, b, 101)
+    ...     bspl = BSpline.basis_element(t[k+j:-k+j])
+    ...     plt.plot(xx, bspl(xx), label=f'j = {j}')
+    >>> plt.legend(loc='best')
+    >>> plt.show()
+
+Here `BSpline.basis_element` is essentially a shorthand for constructing a spline
+with only a single non-zero coefficient. For instance, the ``j=2`` element in
+the above example is equivalent to
+
+    >>> c = np.zeros(t.size - k - 1)
+    >>> c[-2] = 1
+    >>> b = BSpline(t, c, k)
+    >>> np.allclose(b(xx), bspl(xx))
+    True
+
+If desired, a b-spline can be converted into a `PPoly` object using
+`PPoly.from_spline` method which accepts a `BSpline` instance and returns a
+`PPoly` instance. The reverse conversion is performed by the
+`BSpline.from_power_basis` method. However, conversions between bases is best
+avoided because it accumulates rounding errors.
```

Comment:
After reading this, I'm guessing that this is the main point about why to use b-splines:

> The b-spline basis generally more computationally stable then the power basis

Maybe show an example of that? What are the downsides (computation time?)?

Reply:
Don't have a handly example off the bat, not sure there is one without going into *gory* details of what Nikolay called *the shady world of b-splines* when he wrote the CubicSpline code :-). Will think of ways of rephrasing. Maybe need to stress that the localized nature of the basis  often makes linear algebra banded.